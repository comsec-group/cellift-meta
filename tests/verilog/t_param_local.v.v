module t (
	a,
	y
);
	input [1:0] a;
	output wire [3:0] y;
	Test #(.C(2)) test(
		.a(a),
		.y(y)
	);
endmodule
module Test (
	a,
	y
);
	parameter C = 3;
	localparam O = 1 << C;
	input [C - 1:0] a;
	output reg [O - 1:0] y;
	initial begin
		if (O != 4)
			$stop;
		$write("*-* All Finished *-*\n");
	end
endmodule
