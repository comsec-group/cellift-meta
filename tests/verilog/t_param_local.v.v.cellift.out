/* Generated by Yosys 0.14+42 (git sha1 04316fe0d, clang 10.0.0-4ubuntu1 -fPIC -Os) */

module \$paramod\Test\C=s32'00000000000000000000000000000010 (a, y, a_t0, y_t0);
  input [1:0] a;
  wire [1:0] a;
  input [1:0] a_t0;
  wire [1:0] a_t0;
  output [3:0] y;
  wire [3:0] y;
  output [3:0] y_t0;
  wire [3:0] y_t0;
endmodule

module t(a, y, a_t0, y_t0);
  input [1:0] a;
  wire [1:0] a;
  input [1:0] a_t0;
  wire [1:0] a_t0;
  output [3:0] y;
  wire [3:0] y;
  output [3:0] y_t0;
  wire [3:0] y_t0;
  \$paramod\Test\C=s32'00000000000000000000000000000010  test (
    .a(a),
    .a_t0(a_t0),
    .y(y),
    .y_t0(y_t0)
  );
endmodule
